### DISCUSSION TITLE
OutOfMemory Exception

### DISCUSSION THREAD

**Discussion by marcomoret** [IonxSolutions/serilog-sinks-syslog#77] [General]
hi to all, after a big number of events written by syslog using "ILogger log.Information(msg)", the process go in outofmemory. Performance Monitor give me a problem on <<System!Net.Sockets.UdpClient>>.
Where do I make an error and where I can free memory?
thanks

**Comment by colin-anderson** [comment]
Hi, could you please provide a minimal repro? Also, what OS and .NET framework version are you using?

**Reply by marcomoret** [reply]
Hi, follow the repro (I use framework .Net 4.7.2 into a win server 2019 or 2022):

For every block of events:
ILogger sysLogLogger = null;

sysLogLogger = new LoggerConfiguration().MinimumLevel.Information().WriteTo.UdpSyslog(host: <host>, 
    port: <port>, appName: <app>, format: SyslogFormat.RFC3164, 
    facility: (Facility)16, outputTemplate: "").CreateLogger();

Cicle for every event of block:
sysLogLogger.Information(msg);

end of the block of events
Serilog.Log.CloseAndFlush();

and this is reapeted a lot of times to day with tens of events into every block

**Reply by colin-anderson** [reply]
That's more or less what the sample project does, and of course it works fine. In order to investigate, we will need a fully working repro - if you could ZIP and attach a minimal repro project, that would be very helpful, thanks.

**Comment by marcomoret** [comment]
[syslog3.zip](https://github.com/user-attachments/files/17311880/syslog3.zip)

**Comment by marcomoret** [comment]
visual studio 2022 minimal project

**Comment by colin-anderson** [comment]
So, what's happening here is that you are recreating the logger in a tight loop:

```c#
while (true)
{
    Log = new LoggerConfiguration()....  <-- Don't do this in the loop!
}
```

Instead, the logger should be created once, outside of the loop.

**Reply by marcomoret** [reply]
ok, I put the Log object to null after every loop but maybe GC doesn't clean correctly
Now I try your suggest, thanks

**Comment by marcomoret** [comment]
thanks, now it run correctly
